filestore/handlers_base.py

```diff
@@ -0,0 +1,73 @@
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import six
+import logging
+
+
+class HandlerBase(object):
+    """
+    Base-class for Handlers to provide the boiler plate to
+    make them usable in context managers by provding stubs of
+    ``__enter__``, ``__exit__`` and ``close``
+    """
+    specs = set()
```

Comment:
This should be an instance attribute, not a class attribute. Illustration of the problem:

```python
In [1]: class A:
   ...:     specs = set()
   ...:

In [2]: class B(A):
   ...:     pass
   ...:

In [3]: b = B()

In [4]: a = A()

In [5]: b.specs.add('thing')

In [6]: a.specs
Out[6]: {'thing'}
```